src/containers/Heatmap/Heatmap.tsx

```diff
@@ -0,0 +1,198 @@
+                }),
+            );
+        }
+    }, [currentMetric, metrics, dispatch]);
+
+    const fetchData = useCallback(
+        (isBackground: boolean) => {
+            if (!isBackground) {
+                dispatch(setHeatmapOptions({wasLoaded: false}));
+            }
+            dispatch(getTabletsInfo({path}));
+        },
+        [path, dispatch],
+    );
+
+    useAutofetcher(fetchData, [fetchData], autorefresh);
+
+    const onShowTooltip = (...args: Parameters<typeof showTooltip>) => {
+        dispatch(showTooltip(...args));
+    };
+
+    const onHideTooltip = () => {
+        dispatch(hideTooltip);
+    };
+
+    const handleMetricChange = (value: string[]) => {
+        dispatch(
+            setHeatmapOptions({
+                currentMetric: value[0] as IHeatmapMetricValue,
+            }),
+        );
+    };
+    const handleCheckboxChange = () => {
+        dispatch(
+            setHeatmapOptions({
+                sort: !sort,
+            }),
+        );
+    };
+
+    const handleHeatmapChange = () => {
+        dispatch(
+            setHeatmapOptions({
+                heatmap: !heatmap,
+            }),
+        );
+    };
+
+    const renderHistogram = () => {
+        return (
+            <Histogram
+                tablets={tablets}
+                currentMetric={currentMetric}
+                showTooltip={onShowTooltip}
+                hideTooltip={onHideTooltip}
+            />
+        );
+    };
+
+    const renderHeatmapCanvas = () => {
+        const {min, max} = getCurrentMetricLimits(currentMetric, tablets);
+
+        const preparedTablets = tablets.map((tablet) => {
+            const value = currentMetric && Number(tablet.metrics?.[currentMetric]);
+            const colorIndex = getColorIndex(value, min, max);
+            const color = COLORS_RANGE[colorIndex];
+
+            return {
+                ...tablet,
+                color,
+                value,
+                formattedValue: formatNumber(value),
+                currentMetric,
+            };
+        });
+        const sortedTablets = sort
+            ? preparedTablets.sort((x, y) => Number(y.value) - Number(x.value))
+            : preparedTablets;
+
+        return (
+            <div ref={itemsContainer} className={b('items')}>
+                <HeatmapCanvas
+                    tablets={sortedTablets}
+                    parentRef={itemsContainer}
+                    showTooltip={onShowTooltip}
+                    hideTooltip={onHideTooltip}
+                    currentMetric={currentMetric}
+                />
+            </div>
+        );
+    };
+
+    const renderContent = () => {
+        const {min, max} = getCurrentMetricLimits(currentMetric, tablets);
+
+        return (
+            <div className={b()}>
+                <div className={b('filters')}>
+                    <Select
+                        className={b('heatmap-select')}
+                        value={[currentMetric || '']}
```

Comment:
this recreates the value entity every render, changing its reference
judging by the Select sources, this only affects memoization of event handling functions inside the lib
still, not a very reliable approach